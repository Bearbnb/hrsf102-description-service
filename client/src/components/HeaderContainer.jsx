import React from 'react';
import PropTypes from 'prop-types';
import Header from './Header.jsx';
import ListingStats from './ListingStats.jsx';

const HeaderContainer = ({
  guests, bedrooms, beds, baths, name, ownerName, location, livingSpace,
}) => (
  <div>
    <Header
      livingSpace={livingSpace}
      name={name}
      location={location}
    />
    <ListingStats
      guests={guests}
      bedrooms={bedrooms}
      beds={beds}
      baths={baths}
    />
  </div>
);

HeaderContainer.propTypes = {
  guests: PropTypes.number,
  bedrooms: PropTypes.number,
  beds: PropTypes.number,
  baths: PropTypes.number,
  ownerName: PropTypes.string,
  name: PropTypes.string,
  location: PropTypes.string,
  livingSpace: PropTypes.string,

};

HeaderContainer.defaultProps = {
  guests: 0,
  bedrooms: 0,
  beds: 0,
  baths: 0,
  ownerName: 'Charlie P',
  name: 'Cozy thing',
  location: 'San Francisco',
  livingSpace: 'Apartment',
};

export default HeaderContainer;
