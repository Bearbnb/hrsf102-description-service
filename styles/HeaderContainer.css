.wrapper {
  display: grid;
  grid-template-columns: 530px 70px;
  grid-template-rows: 40px 80px 60px
}