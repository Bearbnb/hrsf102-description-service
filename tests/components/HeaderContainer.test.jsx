import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import HeaderContainer from '../../client/src/components/HeaderContainer';

describe('<HeaderContainer />', () => {
  it('should render properly', () => {
    const wrapper = shallow(<HeaderContainer />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
