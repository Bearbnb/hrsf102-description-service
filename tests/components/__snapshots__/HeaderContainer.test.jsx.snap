// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderContainer /> should render properly 1`] = `
<div
  className="wrapper"
>
  <Header
    baths={0}
    beds={0}
    livingSpace="Apartment"
    location="San Francisco"
    name="Cozy thing"
    ownerName="Charlie P"
  />
  <ListingStats
    baths={0}
    bedrooms={0}
    beds={0}
    guests={0}
  />
  <Owner
    ownerName="John Smith"
    ownerPicture="https://s3-us-west-1.amazonaws.com/airbnb-owner-photos/airbnb1.jpg"
  />
</div>
`;
